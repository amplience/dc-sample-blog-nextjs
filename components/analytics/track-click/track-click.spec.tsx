/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import TrackClick from './track-click';
import aa from 'search-insights';

jest.mock('search-insights');

describe('TrackClick', () => {
  test('renders a trackable click element', async () => {
    const component = ShallowRenderer.createRenderer();
    component.render(<TrackClick eventName="Test Button Click" queryId="testId" objectIds={["testObjectId"]} positions={[0]}>
      <button role='button' >Click Me</button>
    </TrackClick>);
    expect(component.getRenderOutput()).toMatchSnapshot();
  });

  test('clicking the button records an event', async () => {
    const component = renderer.create(<TrackClick eventName="Test Button Click" queryId="testId" objectIds={["testObjectId"]} positions={[0]}>
      <button role='button'>Click Me</button>
    </TrackClick>);
    component.root.findByType('div').props.onClick();
    expect(aa.mock.calls).toMatchSnapshot();

  });
});
