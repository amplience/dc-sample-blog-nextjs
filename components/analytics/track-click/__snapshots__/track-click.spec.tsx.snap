// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackClick clicking the button records an event 1`] = `
Array [
  Array [
    "init",
    Object {
      "apiKey": undefined,
      "appId": undefined,
    },
  ],
  Array [
    "clickedObjectIDsAfterSearch",
    Object {
      "eventName": "Test Button Click",
      "index": undefined,
      "objectIDs": Array [
        "testObjectId",
      ],
      "positions": Array [
        0,
      ],
      "queryID": "testId",
    },
  ],
]
`;

exports[`TrackClick renders a trackable click element 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "display": "flex",
    }
  }
>
  <button
    role="button"
  >
    Click Me
  </button>
</div>
`;
